src/fields/field_fe.cc

```diff
@@ -290,11 +311,148 @@ void FieldFE<spacedim, Value>::value_list (const std::vector< Point >  &point_li
 
 
 template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::init_from_input(const Input::Record &rec, const struct FieldAlgoBaseInitData& init_data) {
+	this->init_unit_conversion_coefficient(rec, init_data);
+
+
+	// read mesh, create tree
+    {
+       source_mesh_ = new Mesh( Input::Record() );
+       reader_file_ = FilePath( rec.val<FilePath>("gmsh_file") );
+       ReaderInstances::instance()->get_reader(reader_file_)->read_mesh( source_mesh_ );
+	   // no call to mesh->setup_topology, we need only elements, no connectivity
+    }
+
+	bih_tree_ = new BIHTree( source_mesh_ );
+
+	field_name_ = rec.val<std::string>("field_name");
+
+	map1_ = new MappingP1<1,3>();
+	map2_ = new MappingP1<2,3>();
+	map3_ = new MappingP1<3,3>();
+}
+
+
+
+template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::set_mesh(const Mesh *mesh, bool boundary_domain) {
+	// Mesh can't be set for field which is not initialized.
+	if (field_name_ == "") return;
+
+	dh_ = new DOFHandlerMultiDim( const_cast<Mesh &>(*mesh) );
+	dh_->distribute_dofs(fe1, fe2, fe3);
+    unsigned int ndofs = max(dh_->fe<1>()->n_dofs(), max(dh_->fe<2>()->n_dofs(), dh_->fe<3>()->n_dofs()));
+    dof_indices = new unsigned int[ndofs];
+    // allocate data_
+	unsigned int data_size = dh_->n_global_dofs();
+	Vec v;
+    std::vector<double> data_vec;
+	data_vec.resize(data_size);
+	VecCreateSeqWithArray(PETSC_COMM_SELF, 1, data_size, &(data_vec[0]), &v);
+    data_vec_ = &v;
+    VecGetArray(*data_vec_, &data_);
+}
+
+
+
+template <int spacedim, class Value>
+bool FieldFE<spacedim, Value>::set_time(const TimeStep &time) {
+	// Time can't be set for field which is not initialized.
+	if (field_name_ == "") return false;
```

Comment:
We don't have access to FieldFlags in FieldFE (descendant of FieldAlgoBase). FieldFlags are part of FieldCommon.